lib/ansible/modules/network/onyx/onyx_facts.py

```diff
@@ -196,25 +196,38 @@ def populate(self):
 
 class Interfaces(FactsBase):
 
-    COMMANDS = ['show interfaces ethernet']
+    COMMANDS = ['show version', 'show interfaces ethernet']
 
     def populate(self):
         super(Interfaces, self).populate()
 
-        data = self.responses[0]
+        version_data = self.responses[0]
+        os_version = version_data['Product release']
+        data = self.responses[1]
+
         if data:
-            self.facts['interfaces'] = self.populate_interfaces(data)
+            self.facts['interfaces'] = self.populate_interfaces(data, os_version)
 
-    def populate_interfaces(self, interfaces):
+    def populate_interfaces(self, interfaces, os_version):
         interfaces_dict = dict()
         for if_data in interfaces:
             if_dict = dict()
-            if_dict["MAC Address"] = if_data["Mac address"]
-            if_dict["Actual Speed"] = if_data["Actual speed"]
-            if_dict["MTU"] = if_data["MTU"]
-            if_dict["Admin State"] = if_data["Admin state"]
-            if_dict["Operational State"] = if_data["Operational state"]
-            if_name = if_dict["Interface Name"] = if_data["header"]
+            if os_version >= BaseOnyxModule.ONYX_API_VERSION:
+                for if_name, interface_data in iteritems(if_data):
+                    interface_data = interface_data[0]
+                    if_dict["MAC Address"] = interface_data["Mac address"]
+                    if_dict["Actual Speed"] = interface_data["Actual speed"]
+                    if_dict["MTU"] = interface_data["MTU"]
+                    if_dict["Admin State"] = interface_data["Admin state"]
+                    if_dict["Operational State"] = interface_data["Operational state"]
+                    if_name = if_dict["Interface Name"] = if_name
+            else:
+                if_name = if_dict["Interface Name"] = if_data["header"]
```

Comment:
you have a duplicate code here.
extracting values from json dict should be done once.
Add a function def _extractIfData(interface_data)
that should get all interface data except the interface name